cpp/src/Ice/FileUtil.h

```diff
@@ -125,10 +126,26 @@ namespace IceInternal
         std::string _path;
     };
 
-//
-// Use streamFilename to construct the filename given to std stream classes
-// like ifstream and ofstream.
-//
+    /**
+     * This exception indicates the failure to lock a file.
+     */
+    class ICE_API FileLockException final : public Ice::LocalException
```

Comment:
It's not clear to me we need to define an exception for this situation (vs throwing a plain LocalException). But since we test the type ID in test scripts, I kept it.

Reply:
We never catch it, so we can probably remove it and rework the test.